### PR TITLE
Knollfear/2926 unify delete modal

### PR DESCRIPTION
**Pull request by knollfear**
If you run this locally you can delete objects as you like and just restore them with "docker-compose exec api npm run seed".  
I didn't do the EXACT design, but I think my changes are an improvement and were discussed with Jenn in Slack.

### This pull request changes...

Implements a modal confirmation for deleting almost all object (except for FFY)
Implements a modal confirmation when the user is about to be signed out.

### This is how to verify this change...

Click delete for the desired object and make sure the confirmation modal looks and acts right .

Refer to ticket #2926 for the list of affected objects

### This pull request is ready to merge when...

- [x] Develop has updated automated tests (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] QA has manually tested and created bugs for any issues found (beyond minor fixes)
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### This feature is done when...

- [x] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/actions/editActivity/activityList.js
Comment (tbolt) [inline web/src/actions/editActivity/activityList.js:6]: Based on @thetif comment:
> that's kind of the redux standard so that you can use them in the action and the reducer and avoid typos

we may want to keep it linked to the symbols. maybe we could revisit this in a tech debt ticket and decide if the trade offs are worth it

File: web/src/styles/scss/dialog.scss
Comment (tbolt) [inline web/src/styles/scss/dialog.scss:7]: I pulled this hex color from Figma `#FDB81E` which might be a little different than the default orange. Maybe not though? ha

**Review by tbolt — APPROVED** [review]
Code looks good. I wonder how the `objType` and `title` will fare in the future if we need them for other purposes. Seem fine for now though.

File: web/src/styles/scss/dialog.scss
Comment (knollfear) [reply web/src/styles/scss/dialog.scss:7]: fixed.  I thought I had done that but it got reverted somehow.

File: web/src/actions/editActivity/activityList.js
Comment (knollfear) [reply web/src/actions/editActivity/activityList.js:6]: ok.  replacing symbols in these files.

**Comment by thetif** [comment]
Hi Knoll, the code looks good. I'm still seeing the old window confirm on the following popups though.
- Key Personnel
- State Expense
- State Staff
- Outcome and Metric
- Metric
- Milestone

Also, the Delete APD works, but the modal doesn't close when you click delete.

**Comment by knollfear** [comment]
> Hi Knoll, the code looks good. I'm still seeing the old window confirm on the following popups though.
> 
>     * Key Personnel
> 
>     * State Expense
> 
>     * State Staff
> 
>     * Outcome and Metric
> 
>     * Metric
> 
>     * Milestone
> 
> 
> Also, the Delete APD works, but the modal doesn't close when you click delete.
> 
> I tried doing an empty cache, hard reset, but it didn't get rid of it.

I think I got these fixed.  Delete APD modal is closing for me when I delete.

**Comment by thetif** [comment]
Everything looks good, but the Delete APD modal still isn't closing for me. I tried it in Chrome and Firefox.

File: web/src/containers/StateDashboard.js
Comment (thetif) [inline web/src/containers/StateDashboard.js:222]: I think you want to set setShowDeleteModal to false here.

**Review by thetif — CHANGES_REQUESTED** [review]
update onCancel

**Review by thetif — APPROVED** [review]

**Comment by eStar386** [comment]
Functional Issue: When the user timeout modal appears selecting the stay signed in button doesn't allow the user to stay signed in. The user is trapped on the modal until they elect to sign out and login.

Steps to Reproduce: 
1. Trigger the sign out functionality , occurs after 10 minutes of inactivity. 
2. Select the Stay Signed in Button 

Actual Result: User is trapped on the modal. 

Expected Result: User is able to stay signed in and access the eAPD.

**Review by eStar386 — APPROVED** [review]
Approved the user can delete the following deleting an activity
 deleting a contractor resource
 deleting a non-personnel cost
 deleting an outcome and metric
 deleting an activity milestone
 deleting a state personnel entry
 deleting a key personnel entry
 deleting an APD
 deleting a FFY 
 auth time out

**Comment by beparticular** [comment]
@knollfear I'm still getting the old browser dialogues on this

**Comment by beparticular** [comment]
@knollfear 
I think that these may have gotten implemented with the existing language that was in the system for the browser dialog. 
The following modals need to be updated to match the text in the Figma file attached to the original issue. Including headings, body text, and buttons. 
This is the figma file - https://www.figma.com/file/6eVvo7JjvXiovGTR4BoVgK/CMS-eAPD-2020?node-id=2282%3A21333
- [ ] FFY
- [ ] Key Personnel 
- [ ] Activity 
- [ ] Outcomes
- [ ] Metrics
- [ ] Milestone
- [ ] State Staff (formerly Personnel) 
- [ ] Other State Expense (formerly Non personnel cost)
- [ ] Private Contractor (formerly Contractor resource) 
- [ ] APD 
- [ ] Your session is expiring

I'm not sure this was specified or decided on, so @jeromeleecms can you weigh in on this one? @akuas and I both thought it would be best to go in this direction. 
When someone adds a new item, and they hit done without naming that item - let's default to the name of the section in the second line question. So for instance instead of the modal saying "Do you want to delete Outcome not specified?" It would say "Do you want to delete this Outcome and Metrics?" 
The following modals need to be updated to reflect this:
- [ ] Key Personnel 
- [ ] Private Contractor Cost
- [ ] Outcomes
- [ ] Milestones (only shows "" if there is no title entered) 

The following additional updates need to be made: 
- [ ] "This action cannot be undone" should not have a line break from the rest of the text. 
- [ ] Unable to save changes modal - right now I'm seeing a plain white modal, needs the styling update
- [ ] FFY is still showing the old dialog box
- [ ] Outcomes is still showing quotes around the name in the second line question 
- [ ] Milestones is still showing quotes around the name in the second line question

**Comment by knollfear** [comment]
> @knollfear
> I think that these may have gotten implemented with the existing language that was in the system for the browser dialog.
I implemented the words we agreed upon in the ticket that made all of these standard.  If they are all just slightly different the complexity goes up a lot because now I need a custom delete dialog for each one which introduces a lot of room for error and confusion.

> When someone adds a new item, and they hit done without naming that item - let's default to the name of the section in the second line question. So for instance instead of the modal saying "Do you want to delete Outcome not specified?" It would say "Do you want to delete this Outcome and Metrics?"
> The following modals need to be updated to reflect this:
> 
>     * [ ]  Key Personnel
This is updated
>     * [ ]  Private Contractor Cost
I think this is updated 
>     * [ ]  Outcomes
This is updated
>     * [ ]  Milestones (only shows "" if there is no title entered)
This is updated
> 
> The following additional updates need to be made:
> 
>     * [ ]  "This action cannot be undone" should not have a line break from the rest of the text.
This is updated
>     * [ ]  Unable to save changes modal - right now I'm seeing a plain white modal, needs the styling update
This has been broken out into #2997
>     * [ ]  FFY is still showing the old dialog box
This has been broken out to #2996 
>     * [ ]  Outcomes is still showing quotes around the name in the second line question
This is updated
>     * [ ]  Milestones is still showing quotes around the name in the second line question
This is updated

**Comment by jeromeleecms** [comment]
@knollfear @beparticular @akuas Overall this looks good - there are couple areas I want to verify with you before I approve this:

1) The question mark in the updated modals are missing. But acknowledging that if Jenn is good with that, I'm fine as well.
2) In Delete Activity Milestone Modal, the title reads "Delete Milestone" instead of "Delete Activity Milestone"

I didn't see a response to Jenn's questions/comments, so just stirring the pot to make sure we can close this out.

**Comment by beparticular** [comment]
@jeromeleecms @knollfear 
The milestone modal is correct according to the Figma. 
I could go either way on the question mark tbh
@akuas do you have strong feelings?

**Comment by akuas** [comment]
@knollfear Just to keep a record, we should keep the question mark on the header (Delete Private Contractor?).

**Comment by knollfear** [comment]
> @knollfear Just to keep a record, we should keep the question mark on the header (Delete Private Contractor?).

I have added these back in.  Please let me know if I can merge and close this now.

**Comment by beparticular** [comment]
@knollfear I'm having trouble accessing the test environment. can you check on this and I'll get right on the final review of this issue?

**Comment by knollfear** [comment]
> @knollfear I'm having trouble accessing the test environment. can you check on this and I'll get right on the final review of this issue?

I think this needed to be redeployed when some other things changed.  I am able to log in to the current deployment.

**Review by NAretakis — APPROVED** [review]
This is good to go! thanks.

**Pull request merged**
